filebeat/harvester/log.go

```diff
@@ -99,6 +99,13 @@ func (h *Harvester) Harvest() {
 			event.Text = &text
 			event.JSONFields = message.Fields
 			event.EventMetadata = h.config.EventMetadata
+			if message.Fields != nil {
```

Comment:
The change I made now which leads to the expected result can be found here: https://github.com/elastic/beats/pull/2279/commits/b1a1bb03f2c7056f0eb0cf1c6adf8eb14a58f0d0 So far it seems it doesn't have any other side effects.